scripts/goalie-labels.js

```diff
@@ -16,6 +16,60 @@
 
 const Codeowners = require('codeowners');
 
+const getRepoEvents = async ({ github, context, pull_number }) => {
+  const commits = await github.paginate(github.rest.pulls.listCommits, {
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    pull_number,
+  });
+
+  const reviews = await github.paginate(github.rest.pulls.listReviews, {
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    pull_number,
+  });
+
+  const pullComments = await github.paginate(
+    'GET /repos/{owner}/{repo}/pulls/{pull_number}/comments',
+    {
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      pull_number,
+    },
+  );
+
+  const comments = await github.paginate(github.rest.issues.listComments, {
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    issue_number: pull_number,
+  });
+
+  const events = [
+    ...reviews.map(({ user, submitted_at }) => ({
+      user,
+      updated_at: submitted_at,
+      type: 'review',
+    })),
+    ...commits.map(({ commit, author }) => ({
+      user: author,
+      updated_at: commit.author.date,
+      type: 'commit',
+    })),
+    ...pullComments.map(({ user, updated_at }) => ({
+      user,
+      updated_at,
+      type: 'pr_comment',
+    })),
+    ...comments.map(({ user, updated_at }) => ({
+      user,
+      updated_at,
+      type: 'pr_comment',
+    })),
+  ];
+
+  return events.sort((a, b) => new Date(a.updated_at) - new Date(b.updated_at));
```

Comment:
i guess this normally is perfectly fine in practice, but makes me wonder if there's some perfectly unique ordering key to use instead :) like an autoincrement event id or something

well, no matter, this is fine